lib/jxl/dec_frame.h

```diff
@@ -173,28 +173,21 @@ class FrameDecoder {
   }
 
   // Sets the pixel callback or image buffer where the pixels will be decoded.
-  // This is not supported for all images. If it succeeds, HasRGBBuffer() will
-  // return true.
-  // If it does not succeed, the image is decoded to the ImageBundle passed to
-  // InitFrame instead.
   //
   // @param undo_orientation: if true, indicates the frame decoder should apply
   // the exif orientation to bring the image to the intended display
   // orientation.
   void SetImageOutput(const PixelCallback& pixel_callback, void* image_buffer,
-                      size_t xsize, size_t ysize, JxlPixelFormat format,
-                      bool unpremul_alpha, bool undo_orientation) const {
-    dec_state_->pixel_callback = pixel_callback;
-    dec_state_->image_buffer = image_buffer;
+                      size_t image_buffer_size, size_t xsize, size_t ysize,
+                      JxlPixelFormat format, bool unpremul_alpha,
+                      bool undo_orientation) const {
     dec_state_->width = xsize;
     dec_state_->height = ysize;
```

Comment:
maybe silly question, but how do these relate to image metadata / frame header sizes?

Reply:
They are image (or preview) dimensions, unless coalescing is disabled, in which case they are frame dimensions. The sizes are calculated by GetCurrentDimensions in decode.cc